### ISSUE TITLE
How to define KeyFunctions command

### ISSUE THREAD

**Issue by doggy8088** [mintty/mintty#845]
I wasn't able to output a command execution result by setting the following `KeyFunctions` setting with `Ctrl-Shift-d` shortcut in the mintty config file.

```sh
KeyFunctions=d:'echo -n `date`';
```

What is the correct format to setup `KeyFunctions`?

**Comment by mintty** [comment]
The example in the manual page (`man mintty`) reads like this:
```
KeyFunctions=d:`echo -n `date``
```
Unfortunately, in the online manual page, the outer backquotes (that trigger external command invocation) were incorrectly transformed into typographic quotes which you turned into single quotes. I'll try to fix the HTML manual page generation. Within the external command, there are additional quotes (around `date`); don't worry, this unusual implicit nesting works.

**Comment by Biswa96** [comment]
@mintty Gituub pages now also accepts mark down pages.

**Comment by mintty** [comment]
I meant http://mintty.github.io/mintty.1.html which is HTML format generated with groff.

**Comment by mintty** [comment]
Did you maybe happen to append a Windows newline (CRLF)? Add a trailing ";" to the line just to make sure the `d` definition is properly terminated.
Other modifications to that config file are effective, to make sure?

**Comment by mintty** [comment]
Oh, actually from your paths I notice it's wsltty, not Cygwin mintty. That will be a problem.
It cannot reach out to WSL `echo` in this case. It will try to invoke `echo` and `date` from its own minimal Cygwin installation, where they aren't installed, of course...

**Comment by doggy8088** [comment]
I see.  If I run `%LocalAppData%\wsltty\bin\mintty.exe` standalone, I got this.

![image](https://user-images.githubusercontent.com/88981/52132848-0815f380-267b-11e9-9f15-38747af37316.png)

I think that should be the root cause. 

Is it possible to use `KeyFunctions` with wsltty?

**Comment by mintty** [comment]
There are some tricks you could try, but not really supported:
* Get yourself a Cygwin installation and copy the tools you need, plus all their dependencies (libraries...) into the wsltty bin directory.
* Call a non-Cygwin tool by giving its full pathname, in Windows syntax. This could be a Windows command-line tool or maybe even a WSL tool (probably to be invoked with the wsl.exe launcher). This is all untested stuff...

**Comment by avih** [comment]
> Is it possible to use `KeyFunctions` with wsltty?

> There are some tricks you could try, but not really supported: ...

Correct me if I'm wrong, but the answer should be: Yes, except when the action is a command (using backticks), at which case, for it to work, wsltty should be able to find that command on the system. Keep in mind that wsltty does not have access to the full cygwin environment commands (like `bash`, `echo`, etc).

Would that be correct?

**Comment by doggy8088** [comment]
> * Get yourself a Cygwin installation and copy the tools you need, plus all their dependencies (libraries...) into the wsltty bin directory.

Tried with no luck.

**Comment by mintty** [comment]
@avih,
> > Is it possible to use KeyFunctions with wsltty?

You are correct, I had overlooked this because we only discussed the external command option here before. Internal functions and string insertion work alike, of course.

**Comment by mintty** [comment]
> Tried with no luck.

Actually, the external command is run via `popen` which needs /bin/sh, so that must be made available too.
For most purposes, it should be sufficient to reuse the shell that is already installed:
cd "$LOCALAPPDATA/wsltty/bin"; ln dash sh

Also, the files to copy must be from a Cygwin-64 installation. They can be found out with `ldd`:
```
cd /bin
ldd echo
ldd date
cp echo date cygintl-8.dll cygiconv-2.dll "$LOCALAPPDATA/wsltty/bin/"
```
and already it works :)

**Comment by mintty** [comment]
I had the idea, if you prefer to invoke a WSL command, something like
```
KeyFunctions=w:`wsl uname`
```
should work. I don't know why it doesn't and gives me a "D" instead.

**Comment by doggy8088** [comment]
@mintty I tried this one.

```md
KeyFunctions=d:`wsl echo `date``
```

The output in WSLtty is weird:

![image](https://user-images.githubusercontent.com/88981/52569520-55008380-2e4c-11e9-86bc-adf011603cbd.png)
